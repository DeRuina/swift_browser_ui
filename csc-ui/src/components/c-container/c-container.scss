:host {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 1200px;
}

:host(.padding) {
  padding: 24px;
}
