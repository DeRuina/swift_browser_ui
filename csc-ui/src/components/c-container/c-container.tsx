import { Component, Host, h, Prop } from '@stencil/core';
/**
 * Container component for holding current page content
 * @group Layout
 * @slot - Content to be centered within a page
 */
@Component({
  tag: 'c-container',
  styleUrl: 'c-container.scss',
  shadow: true,
})
export class CContainer {
  /**
   * Maximum width of container in pixels
   */
  @Prop() width: number;

  render() {
    let style = {};

    if (this.width > 0) {
      style = { 'max-width': `${this.width}px` };
    }

    return (
      <Host style={style}>
        <slot></slot>
      </Host>
    );
  }
}
