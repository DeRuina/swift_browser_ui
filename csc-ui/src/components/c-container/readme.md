# c-container



<!-- Auto Generated Below -->


## Overview

Container component for holding current page content

## Properties

| Property | Attribute | Description                          | Type     | Default     |
| -------- | --------- | ------------------------------------ | -------- | ----------- |
| `width`  | `width`   | Maximum width of container in pixels | `number` | `undefined` |


## Slots

| Slot | Description                          |
| ---- | ------------------------------------ |
|      | Content to be centered within a page |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
