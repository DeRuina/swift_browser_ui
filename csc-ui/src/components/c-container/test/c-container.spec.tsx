import { newSpecPage } from '@stencil/core/testing';
import { CContainer } from '../c-container';

describe('c-container', () => {
  it('renders default html', async () => {
    const page = await newSpecPage({
      components: [CContainer],
      html: `<c-container>
      </c-container>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders custom width', async () => {
    const page = await newSpecPage({
      components: [CContainer],
      html: `<c-container width="500">
      </c-container>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
