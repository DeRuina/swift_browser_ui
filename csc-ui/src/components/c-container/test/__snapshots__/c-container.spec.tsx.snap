// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-container renders custom width 1`] = `
<c-container width="500" style="max-width: 500px;">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
</c-container>
`;

exports[`c-container renders default html 1`] = `
<c-container>
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
</c-container>
`;
